sdk/azidentity/chained_token_credential.go

```diff
@@ -78,4 +85,14 @@ func createChainedErrorMessage(errList []CredentialUnavailableError) string {
 	return msg
 }
 
+func formatError(err error, errList []CredentialUnavailableError) error {
+	var authFailed AuthenticationFailedError
+	if errors.As(err, &authFailed) {
+		err = fmt.Errorf("Authentication failed:\n%s\n%s"+createChainedErrorMessage(errList), err)
+		authErr := newAuthenticationFailedError(err, authFailed.RawResponse())
+		return authErr
+	}
+	return err
+}
```

Comment:
I abstracted this away so I could re-use it. There are many ways to do this! So please give me feedback. Part of the goal of this PR is to understand how to do things for the Go SDK.

Reply:
Given that this is specific to ChainedTokenCredential, and it's only used in one place, I don't think there's much value in splitting this out.  At the very least, it should be a method on ChainedTokenCredential due to it specificity.

Reply:
That makes sense! — what about a function within the `GetToken` method? `formatError := func …`? would that be ok? @jhendrixMSFT

Reply:
Sure, but is it used in more than one place?  If the goal is to reduce duplicated code that's great, but I don't see it in this PR.

Reply:
Oh I see it now.  Yeah use a local func.

Reply:
on my way!